src/sage/modular/hypergeometric_motive.py

```diff
@@ -1753,11 +1793,23 @@ def euler_factor(self, t, p, cache_p=False):
             raise ValueError('p not prime')
         if not all(x.denominator() % p for x in self._alpha + self._beta):
             raise NotImplementedError('p is wild')
-        if (t.numerator() * t.denominator() % p == 0 or (t - 1) % p == 0):
+        e = t.valuation(p)
+        if (t.numerator()*t.denominator() % p == 0):
             raise NotImplementedError('p is tame')
-        # now p is good
-        d = self.degree()
-        bound = d // 2
+        # now p is good, or p is tame and t is a p-adic unit
+        if (t-1) % p == 0:
+            d = self.degree()-1
+            bound = d
+            if deg is not None:
+                bound = min(deg, bound)
+            deg = bound
+            use_fe = False
+        else:
+            d = self.degree()
+            bound = d // 2
+            if deg is not None:
+                bound = min(deg, bound)
```

Comment:
This could be pulled outside the `if`-`else` statement to avoid a little bit of duplication.